docarray/math/ndarray.py

```diff
@@ -192,3 +192,57 @@ def get_array_rows(array: 'ArrayType') -> Tuple[int, int]:
         raise ValueError
 
     return num_rows, ndim
+
+
+def check_arraylike_equality(x: 'ArrayType', y: 'ArrayType'):
+    """Check if two array type objects are the same with the supported frameworks.
+
+    Examples
+
+    >>> import numpy as np
+        x = np.array([[1,2,0,0,3],[1,2,0,0,3]])
+        check_arraylike_equality(x,x)
+    True
+
+    >>> from scipy import sparse as sp
+        x = sp.csr_matrix([[1,2,0,0,3],[1,2,0,0,3]])
+        check_arraylike_equality(x,x)
+    True
+
+    >>> import torch
+        x = torch.tensor([1,2,3])
+        check_arraylike_equality(x,x)
+    True
+    """
+    x_type, x_is_sparse = get_array_type(x)
+    y_type, y_is_sparse = get_array_type(y)
+
+    same_array = False
+    if x_type == y_type and x_is_sparse == y_is_sparse:
+        if isinstance(x, list):
+            return x == y
+
+        if x.shape == y.shape:
+            if x_type == 'numpy':
+                # Numpy does not support sparse tensors
+                same_array = (x == y).all()
+            elif x_type == 'torch':
+                if x_is_sparse:
+                    same_array = all((x - y).coalesce().values() == 0)
+                else:
+                    same_array = (x == y).numpy().all()
+            elif x_type == 'scipy':
+                same_array = (x - y).sum() == 0
+            elif x_type == 'tensorflow':
+                if x_is_sparse:
+                    same_array = x == y
+                else:
+                    same_array = (x == y).numpy().all()
+            elif x_type == 'paddle':
+                # Paddle does not support sparse tensor on 11/8/2021
+                # https://github.com/PaddlePaddle/Paddle/issues/36697
+                same_array = (x == y).numpy().all()
```

Comment:
im not so sure about this logic here, shouldn't we use package-specific assert equal there?

e.g. 

- for torch: `torch.eq(a, b)`
- for numpy: `numpy.array_equal`
- for scipy: google "test two scipy matrix are equal"
- for tf
- ...